src/app/util/attribute-storage.cpp

```diff
@@ -842,60 +932,60 @@ bool emberAfEndpointIsEnabled(EndpointId endpoint)
     return emberAfEndpointIndexIsEnabled(index);
 }
 
-// bool emberAfEndpointEnableDisable(EndpointId endpoint, bool enable)
-// {
-//     uint8_t index = findIndexFromEndpoint(endpoint,
-//                                           false); // ignore disabled endpoints?
-//     bool currentlyEnabled;
-
-//     if (0xFF == index)
-//     {
-//         return false;
-//     }
-
-//     currentlyEnabled = emAfEndpoints[index].bitmask & EMBER_AF_ENDPOINT_ENABLED;
-
-//     if (enable)
-//     {
-//         emAfEndpoints[index].bitmask |= EMBER_AF_ENDPOINT_ENABLED;
-//     }
-//     else
-//     {
-//         emAfEndpoints[index].bitmask &= EMBER_AF_ENDPOINT_DISABLED;
-//     }
-
-// #if defined(EZSP_HOST)
-//     ezspSetEndpointFlags(endpoint, (enable ? EZSP_ENDPOINT_ENABLED : EZSP_ENDPOINT_DISABLED));
-// #endif
-
-//     if (currentlyEnabled != enable)
-//     {
-//         if (enable)
-//         {
-//             initializeEndpoint(&(emAfEndpoints[index]));
-//         }
-//         else
-//         {
-//             uint8_t i;
-//             for (i = 0; i < emAfEndpoints[index].endpointType->clusterCount; i++)
-//             {
-//                 EmberAfCluster * cluster = &((emAfEndpoints[index].endpointType->cluster)[i]);
-//                 //        emberAfCorePrintln("Disabling cluster tick for ep:%d, cluster:0x%2X, %p",
-//                 //                           endpoint,
-//                 //                           cluster->clusterId,
-//                 //                           ((cluster->mask & CLUSTER_MASK_CLIENT)
-//                 //                            ? "client"
-//                 //                            : "server"));
-//                 //        emberAfCoreFlush();
-//                 emberAfDeactivateClusterTick(
-//                     endpoint, cluster->clusterId,
-//                     (cluster->mask & CLUSTER_MASK_CLIENT ? EMBER_AF_CLIENT_CLUSTER_TICK : EMBER_AF_SERVER_CLUSTER_TICK));
-//             }
-//         }
-//     }
-
-//     return true;
-// }
+bool emberAfEndpointEnableDisable(EndpointId endpoint, bool enable)
+{
+    uint8_t index = findIndexFromEndpoint(endpoint,
+                                          false); // ignore disabled endpoints?
+    bool currentlyEnabled;
+
+    if (0xFF == index)
+    {
+        return false;
+    }
+
+    currentlyEnabled = emAfEndpoints[index].bitmask & EMBER_AF_ENDPOINT_ENABLED;
+
+    if (enable)
+    {
+        emAfEndpoints[index].bitmask |= EMBER_AF_ENDPOINT_ENABLED;
+    }
+    else
+    {
+        emAfEndpoints[index].bitmask &= EMBER_AF_ENDPOINT_DISABLED;
```

Comment:
this is odd ... disabled enum sounds like a bitflag rather than a 'all but enabled' which is how it is used here.

```cpp
emAfEndpoints[index].bitmask &= ~EMBER_AF_ENDPOINT_ENABLED;
```

may be more obvious